Tasks/AzureCLIV2/azureclitask.ts

```diff
@@ -128,7 +128,7 @@ export class azureclitask {
             let escapedCliPassword = cliPassword.replace(/"/g, '\\"');
             tl.setSecret(escapedCliPassword.replace(/\\/g, '\"'));
             //login using svn
-            Utility.throwIfError(tl.execSync("az", `login --service-principal -u "${servicePrincipalId}" -p "${escapedCliPassword}" --tenant "${tenantId}"`), tl.loc("LoginFailed"));
+            Utility.throwIfError(tl.execSync("az", `login --service-principal -u "${servicePrincipalId}" -p "${escapedCliPassword}" --tenant "${tenantId}" --allow-no-subscriptions`), tl.loc("LoginFailed"));
```

Comment:
Similarly for L143. We should not run this command if subscription is not set.

Reply:
Yes, I think I need to wrap this in a conditional. I will only set the subscription if there is a subscription. I think when there is none, the subscription is set to the tenant ID by default.
https://github.com/microsoft/azure-pipelines-tasks/blob/bf7f4cdf248af83e68ab5b40db51588269764ed8/Tasks/AzureCLIV2/azureclitask.ts#L142-L143

Reply:
I've pushed a commit to account for this.